applications/FluidDynamicsApplication/tests/cpp_tests/test_statistics_utilities.cpp

```diff
@@ -0,0 +1,196 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+//
+
+// Project includes
+#include "testing/testing.h"
+#include "includes/model_part.h"
+#include "includes/cfd_variables.h"
+
+#include "fluid_dynamics_application_variables.h"
+#include "custom_utilities/statistics_record.h"
+#include "custom_utilities/statistics_data.h"
+#include "custom_utilities/statistics_utilities.h"
+
+namespace Kratos {
+namespace Testing  {
+
+namespace Internals {
+
+void TestStatisticsUtilitiesInitializeModelPart(
+    ModelPart& rModelPart, double DeltaTime, unsigned int BufferSize) {
+
+    rModelPart.AddNodalSolutionStepVariable(VELOCITY);
+    rModelPart.AddNodalSolutionStepVariable(PRESSURE);
+    rModelPart.SetBufferSize(BufferSize);
+    Properties::Pointer p_properties = rModelPart.pGetProperties(0);
+
+    // Geometry creation
+    rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+    rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+    rModelPart.CreateNewNode(3, 0.0, 1.0, 0.0);
+    rModelPart.CreateNewNode(4, 0.0, 0.0, 1.0);
+    std::vector<ModelPart::IndexType> element_nodes{1, 2, 3, 4};
+    rModelPart.CreateNewElement("QSVMS3D4N", 1, element_nodes, p_properties);
+
+    // Nodal data
+    Element& r_element = *(rModelPart.ElementsBegin());
+    Geometry< Node<3> >& r_geometry = r_element.GetGeometry();
+
+    for (unsigned int i = 0; i < 4; i++) {
+        Node<3>& r_node = r_geometry[i];
+        r_node.FastGetSolutionStepValue(PRESSURE) = 10.0; // * r_node.Id();
+        r_node.FastGetSolutionStepValue(VELOCITY_X) = 1.0;//r_node.Id() + 5.0;
+        r_node.FastGetSolutionStepValue(VELOCITY_Y) = 2.0;//r_node.Id() + 5.0;
+        r_node.FastGetSolutionStepValue(VELOCITY_Z) = 3.0;//r_node.Id() + 5.0;
+    }
+
+    // Element data
+    r_element.SetValue(C_SMAGORINSKY,0.16);
+
+    // ProcessInfo
+    rModelPart.GetProcessInfo().SetValue(OSS_SWITCH,1);
+
+    // Loop starts at 1 because you need one less clone than time steps
+    for (unsigned int i = 1; i < BufferSize; i++) {
+        rModelPart.CloneTimeStep(i * DeltaTime);
+
+/*        for (unsigned int j = 0; j < 4; j++) {
+            Node<3>& r_node = r_geometry[j];
+            r_node.FastGetSolutionStepValue(PRESSURE) += float(i);
+            r_node.FastGetSolutionStepValue(VELOCITY_Y) = r_node.Id() + i;
+        }*/
+    }
+}
+
+} // namespace internals
+
+KRATOS_TEST_CASE_IN_SUITE(StatisticUtilitiesUsage, FluidDynamicsApplicationFastSuite) {
+    ModelPart model_part("TestModelPart");
```

Comment:
This will have conflicts with the recently merged Model (it happened to me in another PR). Is solved by adding
`Model current_model;`
`ModelPart &model_part = current_mode.CreateModelPart("TestModelPart");`